content/blog/using-jina-embeddings-haystack/index.md

```diff
@@ -0,0 +1,198 @@
+
+We preprocess it by removing whitespace and redundant substrings. Then split it into chunks, generate embeddings, and write these embeddings into the `ChromaDocumentStore`.
+
+```python
+from haystack import Pipeline
+
+from haystack.components.fetchers import LinkContentFetcher
+from haystack.components.converters import PyPDFToDocument
+from haystack.components.writers import DocumentWriter
+from haystack.components.preprocessors import DocumentCleaner
+from haystack.components.preprocessors import DocumentSplitter
+from chroma_haystack.retriever import ChromaEmbeddingRetriever
+from haystack.document_stores import DuplicatePolicy
+
+from jina_haystack.document_embedder import JinaDocumentEmbedder
+from jina_haystack.text_embedder import JinaTextEmbedder
+
+fetcher = LinkContentFetcher()
+converter = PyPDFToDocument()
+# remove repeated substrings to get rid of headers/footers
+cleaner = DocumentCleaner(remove_repeated_substrings=True)
+
+# Since jina-v2 can handle 8192 tokens, 500 words seems like a safe chunk size
+splitter = DocumentSplitter(split_by="word", split_length=500)
+
+# DuplicatePolicy.SKIP is optional but helps avoid errors if you want to re-run the pipeline
+writer = DocumentWriter(document_store=document_store, policy=DuplicatePolicy.SKIP)
+
+retriever = ChromaEmbeddingRetriever(document_store=document_store)
+
+document_embedder = JinaDocumentEmbedder(api_key=jina_api_key, model_name="jina-embeddings-v2-base-en")
+
+indexing_pipeline = Pipeline()
+indexing_pipeline.add_component(instance=fetcher, name="fetcher")
+indexing_pipeline.add_component(instance=converter, name="converter")
+indexing_pipeline.add_component(instance=cleaner, name="cleaner")
+indexing_pipeline.add_component(instance=splitter, name="splitter")
+indexing_pipeline.add_component(instance=document_embedder, name="embedder")
+indexing_pipeline.add_component(instance=writer, name="writer")
+
+indexing_pipeline.connect("fetcher.streams", "converter.sources")
+indexing_pipeline.connect("converter.documents", "cleaner.documents")
+indexing_pipeline.connect("cleaner.documents", "splitter.documents")
+indexing_pipeline.connect("splitter.documents", "embedder.documents")
+indexing_pipeline.connect("embedder.documents", "writer.documents")
+
+# This case references Google V Sonos, October 2023
+urls = ["https://cases.justia.com/federal/district-courts/california/candce/3:2020cv06754/366520/813/0.pdf"]
+
+indexing_pipeline.run(data={"fetcher": {"urls": urls}})
+```
+
+## Building the query pipeline
+
+Now the real fun begins. Let's create a query pipeline so we can actually start asking questions. We write a prompt allowing us to pass our documents to the Mixtral-8x7B LLM. Then we initiatialize the LLM via the `HuggingFaceTGIGenerator`.
+
+In Haystack 2.0 `retriever`s are tightly coupled to `DocumentStores`. If we pass in the `retriever` we initialized earlier, this pipeline can access those embeddings we generated, and pass them to the LLM.
```

Comment:
I haven't completely understood this sentence. did you mean this?  
```suggestion
In Haystack 2.0 `retriever`s are tightly coupled to `DocumentStores`. If we pass the document store in the `retriever` we initialized earlier, this pipeline can access those embeddings we generated, and pass them to the LLM.
```

Reply:
ah thanks, good call, that wording wasn't very clear!